src/commands/git-metadata/library.ts

```diff
@@ -0,0 +1,72 @@
+import {newApiKeyValidator} from '../../helpers/apikey'
+import {RequestBuilder} from '../../helpers/interfaces'
+import {upload, UploadOptions, UploadStatus} from '../../helpers/upload'
+import {getRequestBuilder} from '../../helpers/utils'
+import {getCommitInfo, newSimpleGit} from './git'
+import {CommitInfo} from './interfaces'
+
+export const shouldAddSourceCodeIntegration = async (apiKey: string | undefined): Promise<boolean> => {
+  let simpleGit
+  let isRepo
+  try {
+    simpleGit = await newSimpleGit()
+    isRepo = simpleGit.checkIsRepo()
+  } catch {
+    return false
+  }
+
+  // Only enable if the system has `git` installed and we're in a git repo
+  return apiKey !== undefined && isRepo
```

Comment:
Nit: We try to avoid `let` as much as possible, you can use `return` in the `try` instead
```suggestion
  try {
    const simpleGit = await newSimpleGit()
    const isRepo = simpleGit.checkIsRepo()

    // Only enable if the system has `git` installed and we're in a git repo
    return apiKey !== undefined && isRepo
  } catch {
    return false
  }
```